### PR TITLE
add gitea as an oauth2 backend

### PR DESCRIPTION
**Pull request by Bubu**
This is modeled after gitlab/github.

Gitea oauth docs: https://docs.gitea.io/en-us/oauth2-provider/

User api: https://try.gitea.io/api/swagger#/user/userGetCurrent

Tested with weblate authenticating against gitea, can be tried in the wild at https://weblate.bubu1.eu which allows singing in with codeberg.org accounts.

## Types of changes

Please check the type of change your PR introduces:

- [ ] Release (new release request)
- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Code style update (PEP8, lint, formatting, renaming, etc)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Build related changes (build process, tests runner, etc)
- [ ] Other (please describe):

## Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating
the PR. If you're unsure about any of them, don't hesitate to ask. We're here to
help! This is simply a reminder of what we are going to look for before merging
your code._

- [x] Lint and unit tests pass locally with my changes
- [x] I have added tests that prove my fix is effective or that my feature works

## Other information

Any other information that is important to this PR such as screenshots of how
the component looks before and after the change.

### CODE REVIEW

File: social_core/backends/gitea.py [hunk 1]
```diff
@@ -0,0 +1,51 @@
+"""
+Gitea OAuth2 backend, docs at:
+    https://python-social-auth.readthedocs.io/en/latest/backends/gitea.html
+"""
+from .oauth import BaseOAuth2
+
+
+class GiteaOAuth2   (BaseOAuth2):
+    """Github OAuth authentication backend"""
```
Comment (fnetX) [inline social_core/backends/gitea.py:9]:
```suggestion
    """Gitea OAuth authentication backend"""
```

**Comment by Bubu** [comment]
I tried copying the tests from the gitlab backend but I'm unsure how these work currently.

**Comment by nijel** [comment]
Can you please add the tests? These mostly just mock the user details response and test its parsing.

**Comment by Bubu** [comment]
I added tests based on the existing gitlab tests and filled them with the gitea api responses.

**Comment by nijel** [comment]
Thanks! Can you please also add entry to CHANGELOG.md?

**Comment by Bubu** [comment]
Done!

I suspect https://github.com/python-social-auth/social-docs needs this added as well afterwards?

**Comment by nijel** [comment]
Yes, that would be welcome!

**Comment by nijel** [comment]
Merged, thanks for your contribution!

**Pull request merged**